lib/streamlit/app_session.py

```diff
@@ -216,6 +217,28 @@ def _enqueue_forward_msg(self, msg: ForwardMsg) -> None:
         if self._message_enqueued_callback:
             self._message_enqueued_callback()
 
+    def handle_backmsg(self, msg: BackMsg) -> None:
+        """Process a BackMsg."""
+        try:
+            msg_type = msg.WhichOneof("type")
+
+            if msg_type == "rerun_script":
+                self._handle_rerun_script_request(msg.rerun_script)
+            elif msg_type == "load_git_info":
+                self._handle_git_information_request()
+            elif msg_type == "clear_cache":
+                self._handle_clear_cache_request()
+            elif msg_type == "set_run_on_save":
+                self._handle_set_run_on_save_request(msg.set_run_on_save)
+            elif msg_type == "stop_script":
+                self._handle_stop_script_request()
+            else:
+                LOGGER.warning('No handler for "%s"', msg_type)
+
+        except BaseException as e:
+            LOGGER.error(e)
+            self.handle_backmsg_exception(e)
```

Comment:
nit: `handle_backmsg_exception` can be prefixed with an underscore now too since it's only used here